controllers/configurationpolicy_controller.go

```diff
@@ -1371,97 +1399,183 @@ func (r *ConfigurationPolicyReconciler) determineDesiredObjects(
+	if desiredName == "" && objectSelector != nil {
+		// Parse the objectSelector to determine whether it's valid
+		objSelector, err := metav1.LabelSelectorAsSelector(objectSelector)
+		if err != nil {
+			log.Error(err, "Failed to select the resources",
+				"objectSelector", fmt.Sprint(objectT.ObjectSelector.String()))
 
-	for _, ns := range relevantNamespaces {
-		var rawDesiredObject []byte
+			msg := fmt.Sprintf(
+				"Error parsing provided objectSelector in the object-template at index [%d]: %v",
+				index, err,
+			)
 
-		// Process templating only on the first loop if the ObjectNamespace template variable isn't used
-		if tmplResolver != nil && hasTemplate && (desiredObj == nil || needsPerNamespaceTemplating) {
-			r.processedPolicyCache.Delete(plc.GetUID())
+			errEvent := &objectTmplEvalEvent{
+				compliant: false,
+				reason:    "objectSelector error",
+				message:   msg,
+			}
 
-			templateContext := struct{ ObjectNamespace string }{ObjectNamespace: ns}
+			return nil, &scopedGVR, errEvent, err
+		}
 
-			resolvedTemplate, err := tmplResolver.ResolveTemplate(
-				objectT.ObjectDefinition.Raw, templateContext, resolveOptions,
-			)
+		// Has a valid objectSelector, so list the names for each namespace using the objectSelector
+		for ns := range relevantNsNames {
+			filteredObjects, err := r.DynamicWatcher.List(plc.ObjectIdentifier(), objGVK, ns, objSelector)
 			if err != nil {
-				var complianceMsg string
+				log.Error(err, "Failed to fetch the resources",
+					"objectSelector", fmt.Sprint(objectT.ObjectSelector.String()))
 
-				complianceMsg, err := getFormattedTemplateErr(err)
+				msg := fmt.Sprintf(
+					"Error listing resources with provided objectSelector in the object-template at index [%d]: %v",
+					index, err,
+				)
 
 				errEvent := &objectTmplEvalEvent{
 					compliant: false,
-					reason:    reasonTemplateError,
-					message:   complianceMsg,
+					reason:    "objectSelector error",
+					message:   msg,
 				}
 
 				return nil, &scopedGVR, errEvent, err
 			}
 
-			if objectT.RecordDiff == "" && resolvedTemplate.HasSensitiveData {
-				log.V(1).Info(
-					"Not automatically turning on recordDiff due to templates interacting with sensitive data",
-					"objectTemplateIndex", index,
+			// Populate names from objectSelector results
+			for _, res := range filteredObjects {
+				relevantNsNames[ns] = append(relevantNsNames[ns], res.GetName())
+			}
+		}
+	}
+
+	// Fetch and filter objects using provided objectSelector
+	desiredObjects := []*unstructured.Unstructured{}
+
+	// Detect templates
+	hasTemplate := templates.HasTemplate(objectT.ObjectDefinition.Raw, "", true)
+	needsPerNamespaceTemplating := false
+
+	// Detect .ObjectNamespace template context variables
+	if hasTemplate {
+		needsPerNamespaceTemplating = templateHasObjectNamespaceRegex.Match(objectT.ObjectDefinition.Raw)
+	}
+
+	var desiredObj *unstructured.Unstructured
```

Comment:
Can we add comment why this diff from desiredObjects

Reply:
I've added some comments--let me know if they make sense!